README.md

```diff
@@ -123,69 +49,288 @@ GRANT ALL PRIVILEGES ON DATABASE sample TO manager;
 | SENDGRID_FROM_EMAIL               | <sendgridfromemail>  | The e-mail address used for notifications   |
 | LOCAL                             | False                | Tells the site to use external API or mocks |
 
+# Getting Started
+
+## First time here?
+
+The docker image for this branch is created differently from the other branches (`python manage.py runserver 0.0.0.0` is now run inside the image, not as a command on the image in the `docker-compose.yml`). This means that the old containers and images for other branches need to be deleted so the new one can be built.
+
+```
+$ docker ps --all
+CONTAINER ID     IMAGE               COMMAND                  CREATED        STATUS                  PORTS       NAMES
+<container_id>   wagtail-cms_web     "python manage.py ru…"   3 days ago     Exited (0) 3 days ago               wagtail-cms_web_1
+............     .....               "...................."   ..........     .............           27017/tcp   .................
+$ docker rmi -f <container_id>
+$ docker images
+REPOSITORY                            TAG          IMAGE ID       CREATED         SIZE
+wagtail-cms_web                       latest       <image_id>     11 days ago     1.07GB
+...............                       latest       ............   3 weeks ago     ......
+$ docker rm <image_id>
+...
+```
+
+This now means when you run `docker-compose up` docker will rebuild the image using the new commands in the `Dockerfile`, and then build a new container from that
+
+> *REMEMBER:* To run the commands above when switching back to the other branches on the project
+
+## Tests
+
+```
+$ docker container exec -it wagtail-cms_web_1 python manage.py test
+```
+
+> *IMPORTANT:* When fixing an issue write a test first to prove it's broken. Then, and only then, fix the problem
+
+## Configure
 
-### Running the site
+### Secret Key
 
-#### Running without Docker
+```
+$ python3
+>>> import secrets
+>>> chars = 'abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(-_=+)'
+>>> ''.join(secrets.choice(chars) for i in range(50))
+<secret_key>
+>>> exit()
+$ vim docker-compose.yml
+...
+SECRET_KEY: "<secret_key>"
+...
```

Comment:
Having used this snippet to update the `docker-compose.yml` `SECRET_KEY`, it fails with the following:

```bash
√ wagtail-CMS % python3                                                                                                                                               (rob-fixes)wagtail-CMS 11:57:05
Python 3.9.1 (default, Dec 29 2020, 02:24:05) 
[Clang 12.0.0 (clang-1200.0.32.28)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import secrets
>>> chars = 'abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(-_=+)'
>>> ''.join(secrets.choice(chars) for i in range(50))
'&a85k$+l_s-cbwo6zh8)_)pnzk!(4n0l5^p#+!+_uhiycrb$hz'
>>> exit()
√ wagtail-CMS % docker-compose up                                                                                                                                     (rob-fixes)wagtail-CMS 11:58:30
WARNING: The SEARCHAPIHOST variable is not set. Defaulting to a blank string.
WARNING: The WIDGETAPIHOST variable is not set. Defaulting to a blank string.
WARNING: The WIDGETAPIKEY variable is not set. Defaulting to a blank string.
WARNING: The DATASETAPIHOST variable is not set. Defaulting to a blank string.
WARNING: The DATASETAPIKEY variable is not set. Defaulting to a blank string.
WARNING: The STORAGE_ACCOUNT_NAME variable is not set. Defaulting to a blank string.
WARNING: The STORAGEKEY variable is not set. Defaulting to a blank string.
WARNING: The AZURE_ACCOUNT_NAME variable is not set. Defaulting to a blank string.
WARNING: The AZURE_ACCOUNT_KEY variable is not set. Defaulting to a blank string.
WARNING: The AZURE_CONTAINER variable is not set. Defaulting to a blank string.
WARNING: The SENDGRID_API_KEY variable is not set. Defaulting to a blank string.
ERROR: Invalid interpolation format for "environment" option in service "web": "&a85k$+l_s-cbwo6zh8)_)pnzk!(4n0l5^p#+!+_uhiycrb$hz"
```

However, if I were to replace the secret with a simple string such as `hello`, it works,

Reply:
I have added an additional comment to explain this (https://github.com/office-for-students/wagtail-CMS/blob/rob-fixes/README.md#defaults)